### DISCUSSION TITLE
How can I use model trained by gnina-torch to perform dock in gnina?

### DISCUSSION THREAD

**Discussion by RJ-Li** [RMeli/gnina-torch#53] [Q&A] [answered]
Hi, I want to know if there is an easy way to use model trained by gnina-torch to transfer into Caffe-model-weight file, 
or directly use your gnina-torch model to dock (I didn't find it's possible according to your README)?

**Comment by RMeli** [comment, marked as answer]
Hi @RJ-Li, unfortunately `gnina-torch` does not provide docking capabilities. In principle it should be possible to transfer PyTorch weights to Caffe weights (you can see https://github.com/RMeli/gnina-torch/pull/43#issuecomment-1198620659 as an example of how to do the opposite), but I expect this to be rather tedious and I haven't triesd it myself. Maybe @dkoes or @drewnutt have tried and can provide some guidance?

In [GNINA 1.0: molecular docking with deep learning](https://doi.org/10.1186/s13321-021-00522-2) we suggest to use the deep learning scoring function for re-scoring the poses generated with a classical scoring function, which provides a good tradeoff between accuracy and computational cost. Therefore, one possibility, would be to perform docking with GNINA with `--num_modes` set to the same value of `--num_mc_saved`, so that you will output all the generated poses that are usually re-scored internally (you might also want to turn off RMSD filtering). See Figure 1 of the paper for a visual explanation of the two parameters. You can then re-score all the pose outputted by GNINA with `gnina-torch` and retain only the top $N$ ones (where $N$ is what you would have used as `--num_modes`, `9` by default).
